File: Tutorial5-1Dagger-Android-ContributesAndroidInjector/src/main/java/com/example/tutorial5_1dagger_android_contributesandroidinjector/di/AppComponent.java
package com.example.tutorial5_1dagger_android_contributesandroidinjector.di;

import android.app.Application;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;
import dagger.android.support.DaggerApplication;

/**
 * When @ContributesAndroidInjector is used ActivityContributorModule module is added to modules
 */
@Component(modules = {
        AndroidSupportInjectionModule.class,
        AppModule.class,
        ActivityContributorModule.class})

public interface AppComponent extends AndroidInjector<DaggerApplication> {

    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder application(Application application);

        AppComponent build();
    }
}

File: Tutorial5-1Dagger-Android-ContributesAndroidInjector/src/main/java/com/example/tutorial5_1dagger_android_contributesandroidinjector/di/AppComponent.kt
package com.example.tutorial5_1dagger_android_contributesandroidinjector.di

import android.app.Application
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import dagger.android.support.DaggerApplication
import javax.inject.Singleton

/**
 * When @ContributesAndroidInjector is used ActivityContributorModule module is added to modules
 */
@Singleton
@Component(modules = [AndroidSupportInjectionModule::class,
    AppModule::class,
    ActivityContributorModule::class])
interface AppComponent : AndroidInjector<DaggerApplication> {

    @Component.Builder
    interface Builder {

        @BindsInstance
        fun application(application: Application): Builder

        fun build(): AppComponent
    }
}